### PR TITLE
Added optional flow types for period

### PR DESCRIPTION
**Pull request by ambrinriaz**

### CODE REVIEW

File: src/main/java/com/arkvis/orchid/PeriodCalendar.java [hunk 1]
```diff
@@ -23,6 +23,12 @@ public void addPeriod(LocalDate date) {
         dayMap.put(date, day);
     }
 
+    public void addPeriod(LocalDate date, FlowType flowType) {
```
Comment (lodenrogue) [inline src/main/java/com/arkvis/orchid/PeriodCalendar.java:26]:
```
 public void addPeriod(LocalDate date, FlowType flowType) {
        addPeriod(date);
        dayMap.get(date).addPeriod(flowType);
    }
```

File: src/test/java/com/arkvis/orchid/CalendarTest.java [hunk 2]
```diff
@@ -134,4 +134,30 @@ void should_returnCorrectDate_when_gettingNextPeriodDateOnCalendarWithMultipleCy
...
+    @Test
+    void should_returnNoFlowTypes_when_noFlowTypesAddedToPeriodCycle(){
+        LocalDate date = LocalDate.now();
+
+        PeriodCalendar calendar = new PeriodCalendar(new PeriodPredictor());
+        calendar.addPeriod(date);
+
+        Day retrievedDay  = calendar.getDay(date);
+        Period period =   retrievedDay.getPeriod();
+        FlowType flow = period.getFlowType();
```
Comment (lodenrogue) [inline src/test/java/com/arkvis/orchid/CalendarTest.java:147]: Rename FlowType to Flow

File: src/test/java/com/arkvis/orchid/CalendarTest.java [hunk 3]
```diff
@@ -134,4 +134,30 @@ void should_returnCorrectDate_when_gettingNextPeriodDateOnCalendarWithMultipleCy
...
+    void should_returnFlowTypes_when_flowTypesAddedToPeriodCycle(){
+        LocalDate date = LocalDate.now();
+        PeriodCalendar calendar = new PeriodCalendar(new PeriodPredictor());
+        calendar.addPeriod(date, FlowType.LIGHT);
+
+        Day retrievedDay  = calendar.getDay(date);
+        Period retrievedPeriod =   retrievedDay.getPeriod();
+
+        assertEquals(FlowType.LIGHT, retrievedPeriod.getFlowType());
+
```
Comment (lodenrogue) [inline src/test/java/com/arkvis/orchid/CalendarTest.java:161]: Please add tests for all flow types

**Review by lodenrogue — CHANGES_REQUESTED** [review]

File: src/main/java/com/arkvis/orchid/PeriodCalendar.java (hunk 1 above)
Comment (ambrinriaz) [reply src/main/java/com/arkvis/orchid/PeriodCalendar.java:26]: updated.

File: src/test/java/com/arkvis/orchid/CalendarTest.java (hunk 2 above)
Comment (ambrinriaz) [reply src/test/java/com/arkvis/orchid/CalendarTest.java:147]: updated.

File: src/test/java/com/arkvis/orchid/CalendarTest.java (hunk 3 above)
Comment (ambrinriaz) [reply src/test/java/com/arkvis/orchid/CalendarTest.java:195]: updated.

**Review by lodenrogue — APPROVED** [review]

**Pull request merged**